.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: Plugin CI
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+
+env:
+  PHP_CS_FIXER_IGNORE_ENV: 1
+  XDEBUG_MODE: coverage
+
+jobs:
+  tests:
+    name: "Tests ${{ matrix.php-version }} deps ${{ matrix.dependency-versions }}"
+    runs-on: ubuntu-22.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # normal, highest, non-dev installs
+        php-version: [ '8.2' ]
+        dependency-versions: [ 'highest' ]
+        include:
+          # testing lowest PHP version with the lowest dependencies
+  #        - php-version: '8.2'
+  #          dependency-versions: 'lowest'
+
+          # testing dev versions with the highest PHP
+          - php-version: '8.2'
+            dependency-versions: 'highest'
+
+    steps:
+      - name: "Checkout code"
+        uses: "actions/checkout@v2"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+
+      - name: "Composer install"
+        uses: "ramsey/composer-install@v2"
+        with:
+          dependency-versions: "${{ matrix.dependency-versions }}"
+          composer-options: "--prefer-dist --no-progress"
+
+      - name: Run tests
+        run: composer run test
```

Comment:
@Asisyas, I suggest to split this ci.yaml into static tests (phpcs/phpmd/php-cs-fixer/psalm) and phpunit tests as you don't need static analysis for different php versions (and different versions of dependencies). Although plugin supports only php 8.2, it will be better to split it.

Reply:
It's good.